src/main/java/javaslang/collection/Seq.java

```diff
@@ -90,7 +91,9 @@
  * @author Daniel Dietrich
  * @since 1.1.0
  */
-public interface Seq<T> extends Traversable<T>, IntFunction<T> {
+public interface Seq<T> extends Traversable<T>, Function1<Integer, T> {
```

Comment:
Here is the question: should we generate <code>IntFunctionN</code>, <code>LongFunctionN</code> etc. ?